### PR TITLE
⏰  Add tests for date adapters

### PR DESCRIPTION
**Pull request by martinbonnin**
Follow up from https://github.com/apollographql/apollo-kotlin/pull/3997

### CODE REVIEW

File: apollo-adapters/src/jvmTest/kotlin/JavaTimeAdaptersTest.kt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  @Test
+  fun instant() {
+    var instant = JavaInstantAdapter.fromJson("2010-06-01T22:19:44.475Z")
+    assertEquals(1275430784475, instant.toEpochMilli())
+    assertEquals("2010-06-01T22:19:44.475Z", JavaInstantAdapter.toJson(instant))
+
+    instant = JavaInstantAdapter.fromJson("2010-06-01T23:19:44.475+01:00")
+    assertEquals(1275430784475, instant.toEpochMilli())
```
Comment (BoD) [inline apollo-adapters/src/jvmTest/kotlin/JavaTimeAdaptersTest.kt:42]: Is it really true that `2010-06-01T22:19:44.475Z` and `2010-06-01T23:19:44.475+01:00` are the same instant `1275430784475`? Or does that mean that the offset is also ignored when parsing?

Comment (martinbonnin) [reply apollo-adapters/src/jvmTest/kotlin/JavaTimeAdaptersTest.kt:42]: I think it's true? Second one is 22:19 +1? So both are the same milliseconds offset from epoch (dates are hard...)

Comment (BoD) [reply apollo-adapters/src/jvmTest/kotlin/JavaTimeAdaptersTest.kt:42]: So sorry! Didn't see the `22` vs `23`. Yes it makes sense 👍 .

Too bad about the lost timezone on the writing side, but I don't think it should cause too much trouble.

**Review by BoD — APPROVED** [review]
👍

File: apollo-adapters/src/jvmTest/kotlin/JavaTimeAdaptersTest.kt (hunk 1 above)
Comment (martinbonnin) [reply apollo-adapters/src/jvmTest/kotlin/JavaTimeAdaptersTest.kt:42]: I think it is fine. This is all used to convey "instant" so shouldn't be used to carry a future local date (I have no idea how one would do that)

**Pull request merged**